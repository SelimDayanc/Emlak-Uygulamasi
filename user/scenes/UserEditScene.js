/**
 @flow
 */
import React, {Component, PropTypes} from 'react';
import {
  Dimensions,
  Image,
  ScrollView,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  View,
} from 'react-native';
import colors from '../../common/colors';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Separator from '../../components/Separator';

export default class UserEditScene extends Component {
  static propTypes = {
    user: PropTypes.object.isRequired,
    pickImage: PropTypes.func.isRequired,
    onFieldChange: PropTypes.func.isRequired,
    name: PropTypes.string,
    image: PropTypes.string,
    description: PropTypes.string,
    address: PropTypes.string,
    uploaded: PropTypes.bool.isRequired,
  };

  render() {
    const {
      user,
      uploaded,
      pickImage,
      onFieldChange,
      name,
      image,
      description,
      address,
    } = this.props;

    return (
      <ScrollView style={styles.container}>

        {uploaded
          ? <Image source={{uri: image}} style={styles.logo} />
          : user.image
              ? <Image source={{uri: user.image}} style={styles.logo} />
              : <FontAwesome
                  name="picture-o"
                  color="white"
                  size={200}
                  style={styles.emptyImageIcon}
                />}

        <View style={styles.editIconWrapper}>
          <TouchableHighlight onPress={pickImage} underlayColor="transparent">
            <FontAwesome
              name="camera"
              color={colors.darkGrey}
              size={18}
              style={styles.editIcon}
            />
          </TouchableHighlight>

        </View>

        <View style={styles.content}>
          <Text style={styles.label}>
            {user.isCompany ? 'Company Name' : 'Name'}
          </Text>
          <TextInput
            style={styles.textInput}
            defaultValue={user.name}
            onChangeText={text => onFieldChange('name', text)}
            placeholder="Name"
            placeholderTextColor={colors.lightGrey}
          />
          <Separator style={{marginVertical: 20}} />

          {user.isCompany &&
            <View style={{flex: 1}}>

              <Text style={styles.label}>Company Description</Text>
              <TextInput
                style={styles.textInput}
                defaultValue={user.company.description}
                onChangeText={text => onFieldChange('description', text)}
                multiline={true}
                placeholder="Description"
                placeholderTextColor={colors.lightGrey}
              />
              <Separator style={{marginVertical: 20}} />

              <Text style={styles.label}>Company Address</Text>
              <TextInput
                style={styles.textInput}
                defaultValue={user.company.address}
                onChangeText={text => onFieldChange('address', text)}
                placeholder="Address"
                placeholderTextColor={colors.lightGrey}
                multiline={true}
              />

              <Separator style={{marginVertical: 20}} />
            </View>}

        </View>

      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
  },
  content: {
    padding: 20,
    paddingTop: 50,
  },
  username: {
    fontSize: 20,
    fontWeight: '700',
    color: colors.darkGrey,
  },
  label: {
    fontSize: 15,
    fontWeight: '100',
    color: colors.smokeGreyDark,
  },
  logo: {
    height: 200,
    width: Dimensions.get('window').width,
  },
  emptyImageIcon: {
    height: 200,
    backgroundColor: colors.smokeGreyLight,
    textAlign: 'center',
  },
  editIconWrapper: {
    position: 'absolute',
    top: 200,
    right: 15,
    marginTop: -20,
    height: 40,
    width: 40,
    borderRadius: 20,
    zIndex: 1000,
    alignItems: 'center',
    justifyContent: 'center',
    shadowColor: colors.smokeGreyDark,
    shadowOpacity: 0.6,
    shadowOffset: {width: 1, height: 1},
  },
  textInput: {
    height: 30,
    fontSize: 16,
    marginVertical: 5,
  },
});
