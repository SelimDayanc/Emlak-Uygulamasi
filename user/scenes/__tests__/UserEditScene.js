import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import UserEditScene from '../UserEditScene';

test('renders heart', () => {
  const tree = renderer
    .create(
      <UserEditScene
        user={{name: 'zal', email: 'z4ls@live.com'}}
        pickImage={() => {}}
        onFieldChange={() => {}}
        uploaded={false}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
