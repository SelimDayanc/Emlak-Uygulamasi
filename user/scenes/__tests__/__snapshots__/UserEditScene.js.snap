// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heart 1`] = `
<RCTScrollView
  alwaysBounceHorizontal={undefined}
  alwaysBounceVertical={true}
  onContentSizeChange={null}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={undefined}
  onResponderTerminationRequest={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onScrollShouldSetResponder={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  sendMomentumEvents={false}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      },
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollContentView
    collapsable={false}
    removeClippedSubviews={undefined}
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 200,
          },
          Object {
            "backgroundColor": "#e3e7ed",
            "height": 200,
            "textAlign": "center",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "marginTop": -20,
          "position": "absolute",
          "right": 15,
          "shadowColor": "#5d5f63",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.6,
          "top": 200,
          "width": 40,
          "zIndex": 1000,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#484848",
                "fontSize": 18,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "padding": 20,
          "paddingTop": 50,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#5d5f63",
            "fontSize": 15,
            "fontWeight": "100",
          }
        }
      >
        Name
      </Text>
      <TextInput
        defaultValue="zal"
        onChangeText={[Function]}
        placeholder="Name"
        placeholderTextColor="#f0f0f0"
        style={
          Object {
            "fontSize": 16,
            "height": 30,
            "marginVertical": 5,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#e3e7ed",
              "height": 0.5,
            },
            Object {
              "marginVertical": 20,
            },
          ]
        }
      />
    </View>
  </RCTScrollContentView>
</RCTScrollView>
`;
